apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -1076,6 +1069,48 @@ export class InstallManager {
     });
   }
 
+  private static _mergeEnvironmentVariables(
+    baseEnv: NodeJS.ProcessEnv,
+    environmentVariables: { [environmentVariableName: string]: IEnvironmentVariable; } | undefined
+  ): NodeJS.ProcessEnv {
+    const packageManagerEnv: NodeJS.ProcessEnv = baseEnv;
+
+    if (environmentVariables) {
+      for (const envVar in environmentVariables) {
+        if ({}.hasOwnProperty.call(environmentVariables, envVar)) {
+          let setEnvironmentVariable: boolean = true;
+          console.log(`\nProcessing definition for environment variable: ${envVar}`);
+
+          if (process.env[envVar]) {
+            setEnvironmentVariable = false;
+            console.log(colors.yellow(`WARNING: Environment variable already defined on the device:`));
+            console.log(`  Name: ${envVar}`);
+            console.log(`  Value set on the device: ${process.env[envVar]}`);
+            console.log(`  Value set in Rush config: ${environmentVariables[envVar].value}`);
+
+            if (environmentVariables[envVar].override &&
+              environmentVariables[envVar].override === true) {
+              setEnvironmentVariable = true;
+              console.log(colors.yellow(`WARNING: Overriding the environment variable with the value set in Rush config.`));
+            }
+            else {
+              console.log(colors.yellow(`WARNING: Not overriding the value of the environment variable.`));
+            }
+          }
+
+          if (setEnvironmentVariable === true) {
+            console.log(`Setting environment variable for package manager.`);
```

Comment:
This logging will be really verbose. Maybe we should only log the non-warnings if rush is invoked with `--debug`